2018/2018_File_Generation_Features.md

```diff
@@ -0,0 +1,67 @@
+# 2018 File Generation Features
+
+## The FileGenerator Class. 
+The `FileGenerator()` class is located in [`file_generator.py`](https://github.com/cfpb/hmda-test-files/blob/master/2018/python/file_generator.py) and instantiates without any inputs in the function call. The class uses data stored with yaml configuration files in the 
+[`python/configurations/`](https://github.com/cfpb/hmda-test-files/tree/master/2018/python/configurations) directory. The following yaml configuration files can be altered to create HMDA submission test files.  
+
+-[`test_filepaths.yaml`](https://github.com/cfpb/hmda-test-files/blob/master/2018/python/configurations/test_filepaths.yaml) holds the filepath names for storing clean files, edit test files, log files, and edit reports as well as the paths to the TS and LAR schemas. 
+
+-[`geography_data.yaml`](https://github.com/cfpb/hmda-test-files/blob/master/2018/python/configurations/geographic_data.yaml) holds the filepaths for geographic dependency data such as the tract to CBSA file. The configuration includes the column names for the CBSA file and the columns used for file creation. The file also contains a path to a list of zip codes and a dictionary of FIPS state codes. 
+
+-[`clean_file_config.yaml`](https://github.com/cfpb/hmda-test-files/blob/master/2018/python/configurations/clean_file_config.yaml) holds values for the test institution, file length, and file name for generating a clean HMDA submission file for testing. Other variables, such as data ranges can also be set in this file. 
+
```

Comment:
There needs to be some differentiation here between file generator class, configuration and clean file discussion.

This should be broken out from a user perspective. The use of configurations should be a clearly marked section that comes early in the discussion as these are the parameters of interest when using the tools.

Reply:
This has been done.